### PR TITLE
Check rook ceph pods status after a node failure event

### PR DESCRIPTION
**Pull request by yitzhak12**
Creating a new test to automate [BZ-1861021](https://bugzilla.redhat.com/show_bug.cgi?id=1861021). 
The test checks the rook ceph pods status after a node failure event - the pods should be in a "Running" or "Completed" state.

### CODE REVIEW

File: tests/manage/z_cluster/nodes/test_check_pods_status_after_node_failure.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    rook_ceph_pod_names_set = set(get_pod_name_by_pattern("rook-ceph-"))
+    node_pods = get_node_pods(node_name)
+    node_pod_names_set = set([p.name for p in node_pods])
+    rook_ceph_pod_names_not_in_node = list(rook_ceph_pod_names_set - node_pod_names_set)
+
+    return [get_pod_obj(pod_name) for pod_name in rook_ceph_pod_names_not_in_node]
+
+
+@ignore_leftovers
+@tier4a
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_check_pods_status_after_node_failure.py:43]: please create a polarion test case and add the id here

Comment (yitzhak12) [reply tests/manage/z_cluster/nodes/test_check_pods_status_after_node_failure.py:87]: I added a Polarion test case https://polarion.engineering.redhat.com/polarion/#/project/OpenShiftContainerStorage/workitem?id=OCS-2552 and added the id here.

**Review by ebenahar — APPROVED** [review]

**Review by AviadP — COMMENTED** [review]
In Test file, there are some functions that may be used for another cases, please move it to proper infrastructure file (pod / node etc.)

**Review by prsurve — APPROVED** [review]

**Comment by yitzhak12** [comment]
@AviadP I think the functions in the test file aren't so useful in the `pod` file and are more related to this specific test

**Review by ebenahar — DISMISSED** [review]

**Review by AviadP — DISMISSED** [review]

**Review by AviadP — APPROVED** [review]

**Review by ebenahar — APPROVED** [review]

**Pull request merged**